src/fondant/executor.py

```diff
@@ -154,9 +154,8 @@ def _get_component_arguments(spec: ComponentSpec) -> t.Dict[str, Argument]:
             Input and output arguments of the component.
         """
         component_arguments: t.Dict[str, Argument] = {}
-        kubeflow_component_spec = spec.kubeflow_specification
```

Comment:
Suggestions to add this after line 139

```python
 args.__dict__ = {k: v if v != "None" else None for k, v in args.__dict__.items()}
```

Currently we check if types are `optional` by checking the value in the `default` field (if it's None or another value). However, sometimes the value in the default is a string that we want to override with `None`. 

So either we: 

* Add an additional field to the user component spec where they have to specifically mention if a field is optional -> advantage more explicit, but more things for the user to implement 
* Implement the suggestion above, less things for the user to worry about, might lead to an edge case where the user actually wants to have "None" as a string